anap1003/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment2;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+public class BinaryTree<T> {
+    private TreeNode<T> root;
+
+    /**
+     * Adds a node to the tree, according to its position from the root, given in
+     * the form of a String (i.e. "LLL" should add a node in the leftmost position
+     * in the tree, 3 levels deep).
+     * If a node already exists at the targeted position, the adding fails.
+     * @param key
+     * @param positionFromRoot
+     * @return 
+     */
+    public boolean addNode(T key, String positionFromRoot) {
+        if (positionFromRoot.isEmpty()) { // attempt to add root node 
+            if (root == null) { //no root node in the tree
+                root = new TreeNode(key);
+                return true;
+            }
+            return false; // root node already exists

```

Comment:
Throwing exceptions instead of returning false would prevent a user from writing tests with invalid Trees. Then you don't rely on the user to check the return values.